bundles/org.openhab.binding.thekeys/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,106 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="thekeys"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<bridge-type id="gateway">
+		<label>TheKeys Gateway</label>
+		<description>Gateway controlling a smartlock</description>
+
+		<config-description>
+			<parameter name="host" type="text" required="true">
+				<context>network-address</context>
+				<label>Host</label>
+				<description>Host or IP address of the gateway</description>
+			</parameter>
+			<parameter name="code" type="text" required="true">
+				<context>password</context>
+				<label>Code</label>
+				<description>Code to access the gateway</description>
+			</parameter>
+			<parameter name="refreshInterval" type="integer" min="1" unit="s">
+				<label>Refresh Interval</label>
+				<description>The refresh interval in seconds which is used to poll the gateway</description>
+				<default>5</default>
+			</parameter>
+			<parameter name="apiTimeout" type="integer" min="1" unit="s">
+				<label>API Timeout</label>
+				<description>The api timeout for each request</description>
+				<default>30</default>
+				<advanced>true</advanced>
+			</parameter>
+		</config-description>
+	</bridge-type>
+
+	<thing-type id="smartlock">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="gateway"/>
+		</supported-bridge-type-refs>
+
+		<label>TheKeys Smartlock</label>
+		<description>A smartlock controlled by the gateway</description>
+
+		<channels>
+			<channel id="status" typeId="lock_status"/>
+			<channel id="batteryLevel" typeId="system.battery-level"/>
+			<channel id="lowBattery" typeId="system.low-battery"/>
+			<channel id="rssi" typeId="rssi"/>
+			<channel id="position" typeId="position"/>
+			<channel id="syncInProgress" typeId="syncInProgress"/>
+			<channel id="lastSync" typeId="lastSync"/>
+			<channel id="lock" typeId="smartlockLock"/>
+		</channels>
+
+		<config-description>
+			<parameter name="lockId" type="integer" required="true">
+				<label>Lock ID</label>
+				<description>Identifier of the smartlock</description>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+	<channel-type id="lock_status">
```

Comment:
Please keep a consistent naming convention for your thing type id.
lock_status vs lastSync
=> so maybe lockStatus

Reply:
According to the [naming convention](https://www.openhab.org/docs/developer/guidelines.html#naming-convention) it should be `lock-status` (all lower-case with hyphen) for channel-types, while parameters are `camelCase` (as is already the case here).

Reply:
I updated the thing-id and parameters according to the naming convention